ambari-logsearch/ambari-logsearch-config-zookeeper/src/main/java/org/apache/ambari/logsearch/config/zookeeper/LogLevelFilterManagerZK.java

```diff
@@ -83,7 +83,8 @@ public void setLogLevelFilters(String clusterName, LogLevelFilterMap filters) th
     for (Map.Entry<String, LogLevelFilter> e : filters.getFilter().entrySet()) {
       String nodePath = String.format("/%s/loglevelfilter/%s", clusterName, e.getKey());
       String logLevelFilterJson = gson.toJson(e.getValue());
-      String currentLogLevelFilterJson = new String(serverCache.getCurrentData(nodePath).getData());
+      ChildData childData = serverCache.getCurrentData(nodePath);
```

Comment:
although that fix the NPE, i'm not sure how valid to call this set znode data when it won't do anything,  I guess the error happened when the /logsearch/<clustername> or even /logsearch had not existed yet. so probably on rest api level we should filter out this requests if znodes are not created yet or enhance the whole config api to have an interface call about that setting the log level data cannot be done yet. (with that rest api could return with something like 429 error code, which would make sense) cc @kasakrisz

Reply:
maybe it would be enough just throw a custom exception with a message that can be handled by logsearch rest api

Reply:
overall this change can be ok, probably would need some investigation around acl / znode creation. on trunk/master we would need a more long term solution